config/provider.go

```diff
@@ -580,57 +420,15 @@ func getClientLogMode(configs configs) (m aws.ClientLogMode, found bool, err err
 	return m, found, err
 }
 
-// LogConfigurationWarningsProvider is an interface for retrieving a boolean indicating whether configuration issues should
-// be logged when encountered when loading from config sources.
-type LogConfigurationWarningsProvider interface {
-	GetLogConfigurationWarnings() (bool, bool, error)
-}
-
-// WithLogConfigurationWarnings implements a LogConfigurationWarningsProvider and returns the wrapped boolean value.
-type WithLogConfigurationWarnings bool
-
-// GetLogConfigurationWarnings returns the wrapped boolean.
-func (w WithLogConfigurationWarnings) GetLogConfigurationWarnings() (bool, bool, error) {
-	return bool(w), true, nil
-}
-
-func getLogConfigurationWarnings(configs configs) (v bool, found bool, err error) {
```

Comment:
Probably still want to keep this as a load option paramater, and its functional option helper. In addition probably want to keep all of the `With...` helpers.